src/commands/utils/clusters.ts

```diff
@@ -102,3 +110,71 @@ export async function getClusterProperties(
         return { succeeded: false, error: `Error invoking ${clusterName} managed cluster: ${ex}` };
     }
 }
+
+export async function determineClusterState(
+    target: AksClusterTreeItem,
+    clusterName: string
+): Promise<string | undefined> {
+    try {
+        const containerClient = new azcs.ContainerServiceClient(target.subscription.credentials, target.subscription.subscriptionId);
+        const clusterInfo = (await containerClient.managedClusters.get(target.resourceGroupName, clusterName));
+
+        if ( clusterInfo.provisioningState !== "Stopping" && clusterInfo.agentPoolProfiles?.every((nodePool) => nodePool.powerState?.code === "Stopped") ) {
+                return ClusterState.Stopped;
+        } else if ( clusterInfo.provisioningState === "Succeeded" && clusterInfo.agentPoolProfiles?.every((nodePool) => nodePool.powerState?.code === "Running") ) {
+                return ClusterState.Started;
+        } else if (clusterInfo.provisioningState === "Stopping") {
+            return ClusterState.Stopping;
+        } else {
+            return ClusterState.Starting;
+        }
+
+    } catch (ex) {
+        vscode.window.showErrorMessage(`Error invoking ${clusterName} managed cluster: ${ex}`);
+        return;
+    }
+}
+
+export async function startCluster(
+    target: AksClusterTreeItem,
+    clusterName: string
+): Promise<Errorable<string>> {
+    try {
+        const containerClient = new azcs.ContainerServiceClient(target.subscription.credentials, target.subscription.subscriptionId);
+        const clusterInfo = (await containerClient.managedClusters.get(target.resourceGroupName, clusterName));
```

Comment:
It looks as if we're duplicating the logic in `determineClusterState` here. I'd love it if we could just call that instead... And if `determineClusterState` is changed to return an errorable, then if it fails we can just return the result of that directly.